File: behavioral/command/python/command.py

# Command interface
class Command:
    def execute(self):
        pass

# Receiver class
class Receiver:
    def action(self):
        print("Receiver: Performing an action.")

# Concrete Command class
class ConcreteCommand(Command):
    def __init__(self, receiver):
        self.receiver = receiver

    def execute(self):
        self.receiver.action()

# Invoker class
class Invoker:
    def __init__(self):
        self._command = None

    def set_command(self, command):
        self._command = command

    def execute_command(self):
        if self._command:
            self._command.execute()

if __name__ == "__main__":
    receiver = Receiver()
    command = ConcreteCommand(receiver)

    invoker = Invoker()
    invoker.set_command(command)
    invoker.execute_command()

File: src/behavioral/command/Command.cs

using System;

// Command interface
public interface ICommand
{
    void Execute();
}

// Receiver class
public class Receiver
{
    public void Action()
    {
        Console.WriteLine("Receiver: Performing an action.");
    }
}

// Concrete Command class
public class ConcreteCommand : ICommand
{
    private Receiver _receiver;

    public ConcreteCommand(Receiver receiver)
    {
        _receiver = receiver;
    }

    public void Execute()
    {
        _receiver.Action();
    }
}

// Invoker class
public class Invoker
{
    private ICommand _command;

    public void SetCommand(ICommand command)
    {
        _command = command;
    }

    public void ExecuteCommand()
    {
        _command.Execute();
    }
}

class Program
{
    static void Main()
    {
        Receiver receiver = new Receiver();
        ICommand command = new ConcreteCommand(receiver);

        Invoker invoker = new Invoker();
        invoker.SetCommand(command);
        invoker.ExecuteCommand();
    }
}
